Feature/Add description of Identity deletion and effects of peer deletion on Relationships (#274)
nmshd/documentation: _docs_integrate/data-model-overview.md

| peer             | `string`                                                                                                                                         | The `address` of the [Identity](#identity) with which you have this Relationship.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | saved only locally                            |

> | peerDeletionInfo | [`PeerDeletionInfo`](#peerdeletioninfo) \| `undefined`                                                                                           | Information about whether the Identity with which you have this Relationship is to be deleted or was deleted, as well as the point in time of that deletion.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | saved only locally                            |

Author: We are currently inconsistent in the documentation as to whether we use "it" or the singular "they" when referring to an Identitiy. 

Let's make this decision independently of this PR, document it in our style guide and apply it consistently in the documentation. The same applies to the "peer".

> | reason          | `"Creation"` \| `"AcceptanceOfCreation"` \| `"RejectionOfCreation"` \| `"RevocationOfCreation"` \| `"Termination"` \| `"ReactivationRequested"` \| `"AcceptanceOfReactivation"` \| `"RejectionOfReactivation"` \| `"RevocationOfReactivation"` \| `"Decomposition"` \| `"DecompositionDueToIdentityDeletion"` | The type of the Relationship operation. <br>{::nomarkdown}<ul><li>Creation: creation of the pending Relationship.</li><li>AcceptanceOfCreation: acceptance of the pending Relationship.</li><li>RejectionOfCreation: rejection of the pending Relationship.</li><li>RevocationOfCreation: revocation of the pending Relationship.</li><li>Termination: termination of the active Relationship.</li><li>ReactivationRequested: request for the reactivation of the terminated Relationship.</li><li>AcceptanceOfReactivation: acceptance of the reactivation of the terminated Relationship.</li><li>RejectionOfReactivation: rejection of the reactivation of the terminated Relationship.</li><li>RevocationOfReactivation: revocation of the reactivation of the terminated Relationship.</li><li>Decomposition: decomposition of the terminated Relationship.</li><li>DecompositionDueToIdentityDeletion: decomposition of the terminated Relationship due to the <a href="https://enmeshed.eu/integrate/delete-identities#side-effects-of-identity-deletion-on-relationships">deletion of the peer</a> to which the Relationship exists.</li></ul></ul>{:/} |         |

Author: We are currently inconsistent in the documentation as to whether we use "it" or the singular "they" when referring to a peer.

Let's make this decision independently of this PR, document it in our style guide and apply it consistently in the documentation. The same applies to the "Identity".

## IdentityDeletionProcess

If you want to [delete your Identity]({% link _docs_integrate/delete-identities.md %}), the actual deletion of all data associated with that [Identity](#identity) will only take place after a certain grace period has ended.
Up until then, the data about the deletion process is stored in an object called IdentityDeletionProcess.
It is possible to [cancel an IdentityDeletionProcess]({% link _docs_use-cases/use-case-transport-cancel-identitydeletionprocess.md %}) that hasn't reached the end of its grace period, yet.
Furthermore, the deletion of an Identity can be triggered via the Backbone Admin UI.
In this case, the Identity has to [approve the IdentityDeletionProcess]({% link _docs_use-cases/use-case-transport-approve-identitydeletionprocess.md %}) in order to start the grace period.

> The IdentityDeletionProcess can also be [rejected]({% link _docs_use-cases/use-case-transport-reject-identitydeletionprocess.md %}) if an Identity does not agree with its deletion triggered via the Backbone Admin UI.

Author: I intentionally use the wording "initiating" when the "InitiateIdentityDeletionProcessUseCase" is called and "triggering" when the Identity deletion is triggered via the Backbone Admin UI to differentiate between these two possibilities. This is particularly useful because both options lead to a different starting situation: In the first case, an IdentityDeletionProcess with the status `"Approved"` is created, while in the second case only an IdentityDeletionProcess with the status `"WaitingForApproval"` is created.

Due to a review comment, I have also changed the wording "**by** Backbone Admin UI" to "**via** Backbone Admin UI" so that it does not seems as if the Backbone Admin UI is an acting entity, but is more clearly recognizable as a tool used by a Backbone administrator.

## IdentityDeletionProcess

> Please note that the feature of triggering the deletion of an Identity via the Backbone Admin UI is currently disabled. For this reason, an IdentityDeletionProcess cannot currently have `"WaitingForApproval"` as `status`. In addition, the use cases for [approving]({% link _docs_use-cases/use-case-transport-approve-identitydeletionprocess.md %}) and [rejecting]({% link _docs_use-cases/use-case-transport-reject-identitydeletionprocess.md %}) IdentityDeletionProcesses are no longer needed for the time being.

Reviewer: ```suggestion
Please note that the feature of triggering the deletion of an Identity via the Backbone Admin UI is currently disabled. For this reason, an IdentityDeletionProcess currently cannot have `"WaitingForApproval"` as `status`. In addition, the use cases for [approving]({% link _docs_use-cases/use-case-transport-approve-identitydeletionprocess.md %}) and [rejecting]({% link _docs_use-cases/use-case-transport-reject-identitydeletionprocess.md %}) IdentityDeletionProcesses are not needed for the time being.
```

Same in other places.
